core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -285,8 +287,20 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         sqp.setDefaultOperator(defaultOperator.toBooleanClauseOccur());
 
         Query query = sqp.parse(queryText);
-        if (minimumShouldMatch != null && query instanceof BooleanQuery) {
-            query = Queries.applyMinimumShouldMatch((BooleanQuery) query, minimumShouldMatch);
+        if (query instanceof BooleanQuery) {
+            BooleanQuery booleanQuery = (BooleanQuery) query;
+            if (booleanQuery.clauses().size() > 1
+                    && ((booleanQuery.clauses().get(0)).getQuery() instanceof BooleanQuery) == false) {
+                // special case for one term query and more than one field: f1:t1 f2:t1 f3:t1
+                // we need to wrap this in additional BooleanQuery so minimum_should_match is applied correctly
+                BooleanQuery.Builder builder = new BooleanQuery.Builder();
+                builder.add(new BooleanClause(booleanQuery, Occur.SHOULD));
+                booleanQuery = builder.build();
+            }
+            if (minimumShouldMatch != null) {
```

Comment:
If there isn't more than a single bq clause, should we just ignore applying this at all anyway?